bin/main

```diff
@@ -1,60 +1,46 @@
 #!/usr/bin/env ruby
 
-puts 'WELCOME TO THIS TIC-TAC-TOE GAME: created by: '
+require_relative '../lib/board'
+require_relative '../lib/game'
+require_relative '../lib/player'
+
+def game_loop(game)
+  while game.still_playing
+    puts "#{game.current_player} take your turn!"
+    puts game.board.display
+    puts 'Select a Square:'
+    square_number = get_valid_square(game)
+    game.board.set_a_square(square_number, game.game_piece)
+    game.game_piece = game.game_piece == 'X' ? 'O' : 'X'
+    wining_status = game.win_draw
+    break if %w[X O TIE].include?(wining_status)
+  end
+  game_conclude(game, wining_status)
+end
+
+def game_conclude(game, wining_status)
+  puts game.board.display
+  winner = game.players[wining_status]
+  puts winner ? "#{winner} wins this round." : "It's a TIE"
+end
+
+def get_valid_square(game)
+  game.board.pick_a_square { puts 'Please choose a valid square, choose between 1-9 and that hasn\'t been selected!' }
+end
+
+puts 'WELCOME TO THIS TIC-TAC-TOE GAME: created by: Daniel & Bello.'
 puts "Press 'ENTER' to begin"
 gets
 
-# Ask for the players names
 puts "\nPlayer 1:"
-player1 = gets.chomp
+player1 = get_player('Player 1')
 
 puts "\nPlayer 2:"
-player2 = gets.chomp
+player2 = make_unique_name(player1, get_player('Player 2'))
 
-# Show what numbers to use to place in a square
 puts 'Please choose a number between 1-9 to select your place!'
 
-# An array for the selections the players have chosen
-board = ['', '', '', '', '', '', '', '', '']
-
-# Created a board layout and the indexes of the players selections
-def display_board(board)
-  %(#{board[0]} | #{board[1]} | #{board[2]}
-----------
-#{board[3]} | #{board[4]} | #{board[5]}
-----------
-#{board[6]} | #{board[7]} | #{board[8]}
-  )
-end
-
-# The default game piece
-game_piece = 'X'
+board = Board.new
```

Comment:
I think it's better to move it into the Game constructor. Because you don't use it into that `bin/main` file. You only use it through `game` object.

Reply:
Thanks, but we prefer to keep it. It's in line with the Single responsibility principle.